torch_dvf/transforms.py

```diff
@@ -9,62 +16,54 @@
+class PointCloudHierarchy(ABC):
+    """Abstract class for nested hierarchy of sub-sampled point clouds. Graph edges connect each coarse-scale point to a cluster of fine-scale points.
     Interpolation from the coarse to the fine scales. For correct batching, "torch_geometric.data.Data.__inc__()" has to be overridden.
 
     Args:
         rel_sampling_ratios (Tuple[float]): relative ratios for successive farthest point sampling
         interp_simplex (str): reference simplex for interpolation ('triangle' or 'tetrahedron')
-        cluster_radii (Tuple[float], optional): radii for spherical clusters, estimated from first seen data if None (default: None)
     """
 
-    def __init__(
-        self,
-        rel_sampling_ratios: Tuple[float],
-        interp_simplex: str,
-        cluster_radii: Optional[Tuple[float]] = None,
-        max_num_neighbors: int = 32,
-    ):
+    def __init__(self, rel_sampling_ratios: Tuple[float], interp_simplex: str):
         self.rel_sampling_ratios = rel_sampling_ratios
         self.interp_simplex = interp_simplex
-        self.max_num_neighbors = max_num_neighbors
-
-        self.cluster_radii_arg = cluster_radii  # for "__repr__()"
-        self.cluster_radii = (
-            cluster_radii if cluster_radii else [None] * len(rel_sampling_ratios)
-        )
-
         self.dim_interp_simplex = {"triangle": 2, "tetrahedron": 3}[interp_simplex]
+        self.subsample_keys = ["pos", "batch"]
 
-    def __call__(self, data: pyg.data.Data) -> Data:
+    def _init_state(self, data: pyg.data.Data) -> Dict[str, torch.Tensor]:
         pos = data.pos
         batch = (
             data.surface_id.long()
             if hasattr(data, "surface_id")
             else torch.zeros(pos.size(0), dtype=torch.long)
         )
 
-        for i, (sampling_ratio, cluster_radius) in enumerate(
-            zip(self.rel_sampling_ratios, self.cluster_radii)
-        ):
-            sampling_idcs = fps(
-                pos, batch, ratio=sampling_ratio
-            )  # takes some time but is worth it
+        return {"pos": pos, "batch": batch}
 
-            if cluster_radius is None:
-                cluster_radius = self.estimate_cluster_radius(
-                    pos, pos[sampling_idcs], batch, batch[sampling_idcs]
-                )
-                self.cluster_radii[i] = cluster_radius
+    def _update_state(
+        self, state: Dict[str, torch.Tensor], sampling_idcs: torch.Tensor
+    ) -> Dict[str, torch.Tensor]:
 
-            pool_target, pool_source = radius(
-                pos,
-                pos[sampling_idcs],
-                cluster_radius,
-                batch,
-                batch[sampling_idcs],
-                max_num_neighbors=self.max_num_neighbors,
+        for key in self.subsample_keys:
+            state[key] = state[key][sampling_idcs]
+        return state
+
+    @abstractmethod
+    def _get_hierarchy(
```

Comment:
I was wondering if "hierarchy" is actually what we get here, because when I think of "hierarchy" I think of all scales combined. Maybe "cluster" is more precise?

Reply:
Agree, will change so